resolvers/locationPin.js

```diff
@@ -69,13 +69,18 @@ const locationPinResolvers = {
       return db.locationPins.findByPk(args.id);
     },
     deleteLocationPin: (parent, args) =>
-      // Return status for destroy
-      // 1 for successful deletion, 0 otherwise
-      db.locationPins.destroy({
-        where: {
-          id: args.id,
-        },
-      }),
+      db.locationPins
+        .destroy({
+          where: {
+            id: args.id,
+          },
+        })
+        .then((isDeleted) => {
+          if (isDeleted === 1) {
+            return args.id;
+          }
+          throw new Error('Deletion failed for location pin ID: ' + args.id);
+        }),
```

Comment:
is there a way to generalize this and pull it out like we did for `utils/validators`? (i know it might not work well with the resource associations but this is a little repetitive)

Reply:
could take this part out `(isDeleted) => {
          if (isDeleted === 1) {
            return args.id;
          }
          throw new Error('Deletion failed for location pin ID: ' + args.id);
        }`

and define something like `successCallback = (isDeleted, entityTypeLabel) => {
          if (isDeleted === 1) {
            return args.id;
          }
          throw new Error(`Deletion failed for ${entityTypeLabel} ID: ${args.id}`);
        }`

Reply:
Your call if you want to do that refactoring - our usage of `destroy` is not super consistent/clean so I'm ok if we leave it this way

Reply:
@petorsong do you want this refactoring before i merge the PR?

Reply:
Nah it's fine, I'm good with merging as is